cpmpy/expressions/globalconstraints.py

```diff
@@ -312,6 +312,146 @@ def value(self):
         return arrval in tab
 
 
+class MDD(GlobalConstraint):
+    """The values of the variables in 'array' correspond to a path in the mdd formed by the transition in 'transitions'
+    Root node is the first one used as a start in the first transition
+    """
+
+    def __init__(self, array, transitions):
+        array = flatlist(array)
+        if not all(isinstance(x, Expression) for x in array):
+            raise TypeError("the first argument of an MDD constraint should only contain variables/expressions")
+        super().__init__("mdd", [array, transitions])
+        self.root_node = transitions[0][0]
+        self.mapping = {}
+        for s, v, e in transitions:
+            self.mapping[(s, v)] = e
+
+    def transition_to_layer_representation(self):
+        arr, transitions = self.args
+        nodes_by_level = [[self.root_node]]
+        transitions_by_level = []
+        tran = transitions
+        for i in range(len(arr)):
+            nodes_by_level.append([])
+            transitions_by_level.append([])
+            remaining_tran = []
+            for ns, v, ne in tran:
+                if ns in nodes_by_level[i]:
+                    if ne not in nodes_by_level[i + 1]:
+                        nodes_by_level[i + 1].append(ne)
+                    transitions_by_level[i].append((ns, v, ne))
+                else:
+                    remaining_tran.append((ns, v, ne))
+            tran = remaining_tran
+        return nodes_by_level, transitions_by_level
+
+    # auxillary method to transform into layered representation (gather all the node by node-layers)
+    def normalize_layer_representation(self, nodes_by_level, transitions_by_level):
```

Comment:
add docstring (explain it more than the comment above, as I didn't get it with that one alone)